mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksView.kt

```diff
@@ -0,0 +1,257 @@
+     */
+    fun invalidate()
+
+    /**
+     * The [LifecycleOwner] to use when making new subscriptions. You may want to return different owners depending
+     * on what state your [MvRxView] is in. For fragments, subscriptions made in `onCreate` should use
+     * the fragment's lifecycle owner so that the subscriptions are cleared in `onDestroy`. Subscriptions made in or after
+     * `onCreateView` should use the fragment's _view's_ lifecycle owner so that they are cleared in `onDestroyView`.
+     *
+     * For example, if you are using a fragment as a MvRxView the proper implementation is:
+     * ```
+     *     override val subscriptionLifecycleOwner: LifecycleOwner
+     *        get() = this.viewLifecycleOwnerLiveData.value ?: this
+     * ```
+     *
+     * By default [subscriptionLifecycleOwner] is the same as the MvRxView's standard lifecycle owner.
+     */
+    val subscriptionLifecycleOwner: LifecycleOwner
+        get() = (this as? Fragment)?.viewLifecycleOwnerLiveData?.value ?: this
+
+    fun postInvalidate() {
+        if (pendingInvalidates.add(System.identityHashCode(this@MavericksView))) {
+            handler.sendMessage(Message.obtain(handler, System.identityHashCode(this@MavericksView), this@MavericksView))
+        }
+    }
+
+    /**
+     * Return a [UniqueOnly] delivery mode with a unique id for this fragment. In rare circumstances, if you
+     * make two identical subscriptions with the same (or all) properties in this fragment, provide a customId
+     * to avoid collisions.
+     *
+     * @param An additional custom id to identify this subscription. Only necessary if there are two subscriptions
+     * in this fragment with exact same properties (i.e. two subscribes, or two selectSubscribes with the same properties).
+     */
+    fun uniqueOnly(customId: String? = null): UniqueOnly {
+        return UniqueOnly(listOfNotNull(mvrxViewId, customId).joinToString("_"))
+    }
+
+    /**
+     * Subscribes to all state updates for the given viewModel.
+     *
+     * @param deliveryMode If [UniqueOnly] when this MvRxView goes from a stopped to started lifecycle a state value
+     * will only be emitted if the state changed. This is useful for transient views that should only
+     * be shown once (toasts, poptarts), or logging. Most other views should use false, as when a view is destroyed
+     * and recreated the previous state is necessary to recreate the view.
+     *
+     * Use [uniqueOnly] to automatically create a [UniqueOnly] mode with a unique id for this view.
+     *
+     * Default: [RedeliverOnStart].
+     */
+    fun <S : MvRxState> BaseMavericksViewModel<S>.onEach(deliveryMode: DeliveryMode = RedeliverOnStart, action: suspend (S) -> Unit) =
+        onEachInternal(subscriptionLifecycleOwner, deliveryMode, action)
+
+    /**
+     * Subscribes to state changes for only a specific property and calls the subscribe with
+     * only that single property.
```

Comment:
Could you mention, that if `action` supports cooperative cancellation, previous block will be cancelled?